src/compile/compile_package.zig

```diff
@@ -752,11 +752,19 @@ pub const PackageEnv = struct {
                 // Get a pointer to the child's env (stored in the modules ArrayList)
                 // This is safe because we don't modify the modules ArrayList during type checking
                 const child_env_ptr = &child.env.?;
-                try others.append(child_env_ptr);
+                try imported_envs.append(child_env_ptr);
             }
         }
 
-        var checker = try Check.init(self.gpa, &env.types, env, others.items, &env.store.regions, module_common_idents);
+        var checker = try Check.init(
+            self.gpa,
+            &env.types,
+            env,
+            imported_envs.items,
+            null, // TODO: Propagate other module envs map from Can
```

Comment:
There's probably a failing test we could write which would reproduce this not being brought through 😄